x/accesscontrol/types/wasm.go

```diff
@@ -0,0 +1,53 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+
+	acltypes "github.com/cosmos/cosmos-sdk/types/accesscontrol"
+)
+
+type WasmMessageInfo struct {
+	MessageType     acltypes.WasmMessageSubtype
+	MessageName     string
+	MessageBody     []byte
+	MessageFullBody []byte
+}
+
+func NewExecuteMessageInfo(fullBody []byte) (*WasmMessageInfo, error) {
+	return newMessageInfo(fullBody, acltypes.WasmMessageSubtype_EXECUTE)
+}
+
+func NewQueryMessageInfo(fullBody []byte) (*WasmMessageInfo, error) {
+	return newMessageInfo(fullBody, acltypes.WasmMessageSubtype_QUERY)
+}
+
+func newMessageInfo(fullBody []byte, messageType acltypes.WasmMessageSubtype) (*WasmMessageInfo, error) {
+	name, body, err := extractMessage(fullBody)
+	if err != nil {
+		return nil, err
+	}
+	return &WasmMessageInfo{
+		MessageType:     messageType,
+		MessageName:     name,
+		MessageBody:     body,
+		MessageFullBody: fullBody,
+	}, nil
+}
+
+// WASM message body is JSON-serialized and use the message name
+// as the only top-level key
+func extractMessage(fullBody []byte) (string, []byte, error) {
+	var deserialized map[string]json.RawMessage
+	if err := json.Unmarshal(fullBody, &deserialized); err != nil {
+		return "", nil, err
```

Comment:
i changed it to return `""` and `fullBody`, which would cause GetWasmAccessOps to use BaseAccessOperations only, and then the wasm module can detect the malformat body and handle it the same way as before